addons/website_embedded/models.py

```diff
@@ -1,22 +1,70 @@
 # -*- coding: utf-8 -*-
 
-from openerp import models, fields
+from openerp import models, fields, api
+
+from pyembed.core import PyEmbed, consumer as oembed_consumer
+from requests.exceptions import RequestException
 
 
 class EmbeddedObject(models.Model):
     _name = 'embedded_object'
+    _order = 'sequence, create_date'
 
     KINDS = [
-        ('elearning', "e-learning"),
+        ('elearning', "e-learning (dla wolontariuszy)"),
+        ('elearning_partners', "e-learning (dla partnerów)"),
         ('biuletyn', "biuletyn wolontariuszy"),
     ]
 
-    name = fields.Char(string=u"Nazwa", required=True)
+    name = fields.Char(string=u"Tytuł", required=True)
     kind = fields.Selection(
         KINDS,
         required=True,
         string=u"Typ",
     )
+    resource_url = fields.Char(
+        string=u"Adres zasobu",
+        required=True,
+        help="np. adres filmu na YouTube lub numeru na issuu.com",
+    )
     embed_code = fields.Text(string="Kod osadzenia", required=True)
     thumbnail_url = fields.Char(string=u"Adres miniaturki", required=True)
     description = fields.Html(string=u"Opis")
+    sequence = fields.Integer(default=1)
+
+    @api.onchange('resource_url')
+    def resource_change(self):
+        error_message = None
+        if self.resource_url:
+            # Discover embed code using oEmbed
+            oembed = PyEmbed()
+            try:
+                oembed_urls = oembed.discoverer.get_oembed_urls(self.resource_url)
+                if oembed_urls:
+                    oembed_response = oembed_consumer.get_first_oembed_response(
+                        oembed_urls,
+                        max_width=800,
+                        max_height=600
+                    )
+                    embed_code = oembed.renderer.render(self.resource_url, oembed_response)
+                    thumbnail_url = getattr(oembed_response, 'thumbnail_url', '')
+                if not (oembed_urls and embed_code and thumbnail_url):
```

Comment:
This is tricky. If `oembed_urls` evaluates to `False`, then `embed_code` and `thumbnail_url` are not defined. Therefore putting `oembed_urls` is for the sake of preventing unbound variable errors. Maybe it is better to explicitly initialize `embed_code` and `thumbnail_url` with `None` (“explicit is better than implicit”)? Also why not reverse the order in `if`, so we can get rid of `not`?